src/Microsoft.Fx.Portability/Analysis/AnalysisEngine.cs

```diff
@@ -221,6 +224,74 @@ public IEnumerable<string> FindUnreferencedAssemblies(IEnumerable<string> unrefe
             }
         }
 
+        public IList<NuGetPackageInfo> GetNuGetPackagesInfo(IEnumerable<string> assemblies, IEnumerable<FrameworkName> targets)
+        {
+            var nugetPackages = new List<NuGetPackageInfo>();
+            foreach (var assembly in assemblies)
+            {
+                ImmutableDictionary<FrameworkName, IEnumerable<NuGetPackageId>> packages = null;
+                var result = _packageFinder.FindPackage(assembly, targets, out packages);
+                if (result)
+                {
+                    foreach (var target in targets)
+                    {
+                        var nuGetPackageInfo = new NuGetPackageInfo(assembly, target, packages.ContainsKey(target) ? packages[target].ToList() : new List<NuGetPackageId>());
+                        nugetPackages.Add(nuGetPackageInfo);
+                    }
+                }
+            }
+            return nugetPackages;
+        }
+
+        // Get assemblies that should be removed (nuget packages exist for all targets)
+        public List<AssemblyInfo> ComputeAssembliesToRemove(IEnumerable<AssemblyInfo> userAssemblies, IEnumerable<FrameworkName> targets, out List<NuGetPackageInfo> nugetPackagesForUserAssemblies)
```

Comment:
Can we split this method into two methods? It appears this method is doing more than "computing NuGet packages to be removed".  It appears to be also returning all suggested NuGet packages for user assemblies.

I'm generally wary of methods that use `out` and return something other than a bool or an Enum because usually they're doing more than one function.

It may be better to:
1. Get all the suggested NuGet packages for user assemblies (returned by the out List<NuGetPackageInfo> nugetpackagesforUserAssemblies)
2. Have a filter method to remove all ones that are not supported by all platforms

Reply:
Changed into two methods